Reduce MSRV to 1.53

GnomedDev opened the pull request:
This is part of reducing the MSRV of nougat to be used potentially in serenity.

GnomedDev committed:
Reduce MSRV to 1.53

GnomedDev committed:
Change references to 1.57

danielhenrymantilla commented:
Hmm, for `macro_rules_attribute` the original high MSRV it has is a smell: I think it's needed for the _ergonomic_ `derive` shadowing. This will require that the integration tests of the ergonomic derive be cfg-gated, so since it's a bit more complex, I'd like to take it from here.

Thanks anyways!

danielhenrymantilla committed:
Handle the lack of `#[derive]` shadowing on <1.57.0

  - this is done by adding a disclaimer on the docs;

  - and having the CI runs cheat a bit and enable the then-unstable
    feature so that it can mimic 1.57.0 behavior

danielhenrymantilla committed:
Next version release candidate

danielhenrymantilla committed:
Version 0.1.2 release
